enviroplus_exporter.py

```diff
@@ -78,28 +90,60 @@
 # Setup Luftdaten
 LUFTDATEN_TIME_BETWEEN_POSTS = int(os.getenv('LUFTDATEN_TIME_BETWEEN_POSTS', '30'))
 
-# Get the temperature of the CPU for compensation
+# Setup Safecast
+SAFECAST_TIME_BETWEEN_POSTS = int(os.getenv('SAFECAST_TIME_BETWEEN_POSTS', '300'))
+SAFECAST_DEV_MODE = os.getenv('SAFECAST_DEV_MODE', 'false') == 'true'
+SAFECAST_API_KEY = os.getenv('SAFECAST_API_KEY', '')
+SAFECAST_API_KEY_DEV = os.getenv('SAFECAST_API_KEY_DEV', '')
+SAFECAST_LATITUDE = os.getenv('SAFECAST_LATITUDE', '')
+SAFECAST_LONGITUDE = os.getenv('SAFECAST_LONGITUDE', '')
+SAFECAST_DEVICE_ID = int(os.getenv('SAFECAST_DEVICE_ID', '226'))
+SAFECAST_LOCATION_NAME = os.getenv('SAFECAST_LOCATION_NAME', '')
+if SAFECAST_DEV_MODE:
+    # Post to the dev API
+    safecast = SafecastPy.SafecastPy(
+        api_key=SAFECAST_API_KEY_DEV,
+        api_url=SafecastPy.DEVELOPMENT_API_URL,
+    )
+else:
+    # Post to the production API
+    safecast = SafecastPy.SafecastPy(
+        api_key=SAFECAST_API_KEY,
+    )
+
+# Setup Blues Notecard
+NOTECARD_TIME_BETWEEN_POSTS = int(os.getenv('NOTECARD_TIME_BETWEEN_POSTS', '600'))
+
+# Setup LC709023F battery monitor
+if DEBUG:
+    logging.info('## LC709023F battery monitor ##')
+try:
+    if DEBUG:
+        logging.info("Sensor IC version: {}".format(hex(sensor.ic_version)))
+    # Set the battery pack size to 3000 mAh
+    sensor.pack_size = PackSize.MAH3000
+    sensor.init_RSOC()
+    if DEBUG:
+        logging.info("Battery size: {}".format(PackSize.string[sensor.pack_sizes]))
+except RuntimeError as exception:
+    logging.error("Failed to read sensor with error: {}".format(exception))
+    logging.info("Try setting the I2C clock speed to 10000Hz")
+
 def get_cpu_temperature():
+    """Get the temperature from the Raspberry Pi CPU"""
     with open("/sys/class/thermal/thermal_zone0/temp", "r") as f:
         temp = f.read()
         temp = int(temp) / 1000.0
-    return temp
+        CPU_TEMPERATURE.set(temp)
 
-def get_temperature(factor):
+def get_temperature(temperature_compensation):
     """Get temperature from the weather sensor"""
-    # Tuning factor for compensation. Decrease this number to adjust the
-    # temperature down, and increase to adjust up
-    raw_temp = bme280.get_temperature()
-
-    if factor:
-        cpu_temps = [get_cpu_temperature()] * 5
-        cpu_temp = get_cpu_temperature()
-        # Smooth out with some averaging to decrease jitter
-        cpu_temps = cpu_temps[1:] + [cpu_temp]
-        avg_cpu_temp = sum(cpu_temps) / float(len(cpu_temps))
-        temperature = raw_temp - ((avg_cpu_temp - raw_temp) / factor)
-    else:
-        temperature = raw_temp
```

Comment:
@sighmon what's your idea about changing this to a more basic approach?

Reply:
@tijmenvandenbrink It wasn't working out linearly across the temperature range for me and also didn't take into account the humidity changes from the extra heat. My simplistic version isn't great either, so I'll keep working on it.